### PR TITLE
[Fix] Fix a vital performance bug when run in graph mode deepray will…

### PR DESCRIPTION
**Pull request by fuhailin**
broadcast at the beginning of every execution

# Description
Broadcast too many times
image

Brief Description of the PR:

Fixes # (issue)

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/deepray/blob/master/deepray/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/deepray/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for Deepray
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: deepray/core/base_trainer.py [hunk 1]
```diff
@@ -708,7 +709,7 @@ def run_customized_training_loop(
     # Training loop starts here.
     self.current_step = self._first_steps = self.optimizer.iterations.numpy()
 
-    self.first_batch = True
+    self.first_batch = tf.Variable(True, trainable=False, dtype=tf.bool, name='first_batch')
```
Comment (MoFHeka) [inline deepray/core/base_trainer.py:712]: Creating variable first_batch under init_scope would be better? Which makes sure this variable was created in eager mode to be compatible with python logical controlling code in any time, and also guarantees a safety from TF tape and other contexts.
```python
with tf.init_scope():
    # Initialization runs with eager execution enabled
    # assert tf.executing_eagerly()
    tf.Variable(True, trainable=False, dtype=tf.bool, name='first_batch')
```

Comment (fuhailin) [reply deepray/core/base_trainer.py:712]: Done.
